qcodes/tests/test_channels.py

```diff
@@ -0,0 +1,221 @@
+        # it's not possible to set via self.instrument.channels.temperature as this is a multi parameter
+        # that currently does not support set.
+
+    def test_add_channel(self):
+        n_channels = len(self.instrument.channels)
+        name = 'foo'
+        channel = DummyChannel(self.instrument, 'Chan'+name, name)
+        self.instrument.channels.append(channel)
+        self.instrument.add_submodule(name, channel)
+
+        self.assertEqual(len(self.instrument.channels), n_channels+1)
+
+        self.instrument.channels.lock()
+        # after locking the channels it's not possible to add any more channels
+        with self.assertRaises(AttributeError):
+            name = 'bar'
+            channel = DummyChannel(self.instrument, 'Chan' + name, name)
+            self.instrument.channels.append(channel)
+        self.assertEqual(len(self.instrument.channels), n_channels + 1)
+
+    def test_add_channels_from_generator(self):
+        n_channels = len(self.instrument.channels)
+        names = ('foo', 'bar', 'foobar')
+        channels = (DummyChannel(self.instrument, 'Chan'+name, name) for name in names)
+        self.instrument.channels.extend(channels)
+
+        self.assertEqual(len(self.instrument.channels), n_channels + len(names))
+
+    def test_add_channels_from_tuple(self):
+        n_channels = len(self.instrument.channels)
+        names = ('foo', 'bar', 'foobar')
+        channels = tuple(DummyChannel(self.instrument, 'Chan'+name, name) for name in names)
+        self.instrument.channels.extend(channels)
+
+        self.assertEqual(len(self.instrument.channels), n_channels + len(names))
+
+    def test_insert_channel(self):
+        n_channels = len(self.instrument.channels)
+        name = 'foo'
+        channel = DummyChannel(self.instrument, 'Chan'+name, name)
+        self.instrument.channels.insert(1, channel)
+        self.instrument.add_submodule(name, channel)
+
+        self.assertEqual(len(self.instrument.channels), n_channels+1)
+        self.assertIs(self.instrument.channels[1], channel)
+        self.instrument.channels.lock()
+        # after locking the channels it's not possible to add any more channels
+        with self.assertRaises(AttributeError):
+            name = 'bar'
+            channel = DummyChannel(self.instrument, 'Chan' + name, name)
+            self.instrument.channels.insert(2, channel)
+        self.assertEqual(len(self.instrument.channels), n_channels + 1)
+
+    @given(setpoints=hst.lists(hst.floats(0, 300), min_size=4, max_size=4))
+    def test_combine_channels(self, setpoints):
+        self.assertEqual(len(self.instrument.channels), 6)
+
+        mychannels = self.instrument.channels[0:2] + self.instrument.channels[4:]
+
+        self.assertEqual(len(mychannels), 4)
+        assert mychannels[0] is self.instrument.A
+        assert mychannels[1] is self.instrument.B
+        assert mychannels[2] is self.instrument.E
+        assert mychannels[3] is self.instrument.F
```

Comment:
Fixed. In general the `self.assert` versions provide better introspection when the tests fails but pytest which we currently use does that on regular asserts too. But anyway lets try to keep this uniform